crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs

```diff
@@ -0,0 +1,673 @@
+use super::{
+    dynamic_dory_structure::{full_width_of_row, index_from_row_and_column, matrix_size},
+    pairings, DynamicDoryCommitment, G1Affine, ProverSetup,
+};
+use crate::{
+    base::{commitment::CommittableColumn, slice_ops::slice_cast},
+    proof_primitive::dory::{offset_to_bytes::OffsetToBytes, pack_scalars::min_as_f},
+};
+use ark_ec::CurveGroup;
+use ark_std::ops::Mul;
+use blitzar::compute::ElementP2;
+use std::sync::Mutex;
+use tracing::{span, Level};
+
+const BYTE_SIZE: u32 = 8;
+
+/// Finds the single packed byte size of all committable columns.
+///
+/// # Arguments
+///
+/// * `committable_columns` - A reference to the committable columns.
+///
+/// # Returns
+///
+/// The single packed byte size of all committable columns.
+fn single_packed_byte_size(committable_columns: &[CommittableColumn]) -> usize {
+    committable_columns
+        .iter()
+        .fold(0, |acc, x| acc + x.column_type().byte_size())
+}
+
+/// Returns the size of the matrix that can hold the longest committable column in a dynamic Dory structure.
+///
+/// # Arguments
+///
+/// * `committable_columns` - A reference to the committable columns.
+/// * `offset` - The offset to the data.
+///
+/// # Returns
+///
+/// A tuple containing the maximum (height, width) needed to store the longest committable column in
+/// a dynamic Dory structure.
+fn max_matrix_size(committable_columns: &[CommittableColumn], offset: usize) -> (usize, usize) {
+    committable_columns
+        .iter()
+        .map(|column| matrix_size(column.len(), offset))
+        .fold((0, 0), |(acc_height, acc_width), (height, width)| {
+            (acc_height.max(height), acc_width.max(width))
+        })
+}
+
+/// Returns a single element worth of bit values for the bit table with offsets needed to handle the
+/// signed columns. Note, the signed bits are handled naively. For each committable column,
+/// a signed bit offset entry is added to the bit table. This means every signed and unsigned
+/// committable column will have an additional byte for the signed offset. Also, multiple redundant
+/// commitments of `1`'s are calculated.
+///
+/// # Arguments
+///
+/// * `committable_columns` - A reference to the committable columns.
+/// * `signed_offset_length` - The length of the signed offset.
+///
+/// # Returns
+///
+/// A vector containing the bit sizes of each committable column with a corresponding offset for
+/// a single entry in the bit table.
+fn populate_single_bit_array_with_offsets(
+    committable_columns: &[CommittableColumn],
+    signed_offset_length: usize,
+) -> Vec<u32> {
+    let mut bit_sizes: Vec<u32> = committable_columns
+        .iter()
+        .map(|column| column.column_type().bit_size())
+        .collect();
+
+    bit_sizes.extend(std::iter::repeat(BYTE_SIZE).take(signed_offset_length));
+
+    bit_sizes
```

Comment:
Put a `use core::iter` at the top. (Avoid `std`.)

```suggestion
    committable_columns
        .iter()
        .map(|column| column.column_type().bit_size())
        .chain(iter::repeat(BYTE_SIZE).take(signed_offset_length))
        .collect()
```

Or, use `itertools::repeat_n(BYTE_SIZE, signed_offset_length)`.